sycl/doc/DeviceGlobal.md

```diff
@@ -0,0 +1,617 @@
+# Implementation design for "device\_global"
+
+This document describes the implementation design for the DPC++ extension
+[SYCL\_EXT\_ONEAPI\_DEVICE\_GLOBAL][1], which allows applications to declare
+global variables in device code.
+
+[1]: <extensions/DeviceGlobal/SYCL_INTEL_device_global.asciidoc>
+
+
+## Requirements
+
+The extension specification document referenced above contains the full set of
+requirements for this feature, but some requirements that are particularly
+relevant to the design are called out here.
+
+The first issue relates to the mechanism for integrating host and device code.
+Like specialization constants, device global variables are referenced in both
+host and device code, so they require some mechanism to correlate the variable
+instance in device code with the variable instance in host code.  The API for
+referencing a device global variable, however, is different from the API for
+specialization constants.  Whereas specialization constants are referenced
+through a templated member function:
+
+```
+sycl::specialization_id<int> spec_var;
+
+void func(sycl::queue q) {
+  q.submit([&](sycl::handler &cgh) {
+    cgh.set_specialization_constant<spec_var>(42);
+    cgh.single_task([=](sycl::kernel_handler kh) {
+      int i = kh.get_specialization_constant<spec_var>();
+    });
+  });
+}
+```
+
+Device global variables, by contrast, are referenced by their address:
+
+```
+sycl::ext::oneapi::device_global<int> dev_var;
+
+void func(sycl::queue q) {
+  int val = 42;
+  q.copy(&val, dev_var).wait();      // The 'dev_var' parameter is by reference
```

Comment:
Perhaps we could have an overload in SYCL for scalars too so it is possible to avoid the `&`?

Reply:
Makes sense.  I think we need to do it for all copy functions in SYCL and not just this one, so I opened a [Khronos issue to track](https://github.com/KhronosGroup/SYCL-Docs/issues/204)